controller/linodemachine_controller.go

```diff
@@ -278,14 +278,36 @@ func (r *LinodeMachineReconciler) reconcileCreate(
 	return ctrl.Result{}, nil
 }
 
+// createInstance provisions linode instance after checking if the request will be within the rate-limits
+// Note: it takes a lock before checking for the rate limits and releases it after making request to linode API or when returning from function
+func (r *LinodeMachineReconciler) createInstance(ctx context.Context, logger logr.Logger, machineScope *scope.MachineScope, createOpts *linodego.InstanceCreateOptions) (*linodego.Instance, error, bool) {
```

Comment:
I think this function could be reworked to avoid extra return values. We should define specific type of errors, and perform checks for those specific errors when the function is called.
```
errInstanceCreate
errRateLimit
```
and remove bool

This will also help in testing by creating tests for specific errors

Reply:
updated code based on suggestion